WebApp/Controllers/HomeController.cs

```diff
@@ -49,36 +43,41 @@ public ActionResult About()
 
         [Authorize]
         public async Task<ActionResult> SendMail()
-        {            
-            // try to get token silently
-            string signedInUserID = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
-            TokenCache userTokenCache = new MSALSessionCache(signedInUserID, this.HttpContext).GetMsalCacheInstance();            
-            ConfidentialClientApplication cca = new ConfidentialClientApplication(clientId, redirectUri,new ClientCredential(appKey), userTokenCache, null);
-            var accounts = await cca.GetAccountsAsync();
-            if (accounts.Any())
+        {
+            // Before we render the send email screen, we use the incremental consent to obtain and cache the access token with the correct scopes
+            IConfidentialClientApplication app = MsalAppBuilder.BuildConfidentialClientApplication();
+            AuthenticationResult result = null;
+            var accounts = await app.GetAccountsAsync();
+            string[] scopes = { "Mail.Send" };
+
+            try
             {
-                string[] scopes = { "Mail.Send" };
+                // try to get token silently
```

Comment:
Maybe change the comment to "try to get an already cached token"

Reply:
Fixed